### ISSUE TITLE
Support using Vault for all secrets

### ISSUE THREAD

**Issue by lkysow** [hashicorp/consul-helm#780]
Would be great to support be able to retrieve all secret material: ACL tokens, TLS certs, etc. from Vault.

**Comment by shellfu** [comment]
This would make upgrades/rebuilds immensely easier. If any code contribution is needed for this I will dedicate the time if so desired.

**Comment by david-yu** [comment]
@shellfu Could you give us more details on how that would upgrades/rebuilds easier? Do you currently also use Vault today?

**Comment by shellfu** [comment]
Yes, we currently use Vault to manage the majority of our secrets. Having Vault manage the certificates in the context of K8s would be ideal as rotations could be managed in Vault vs on Kube.
